src/ServiceProvider.php

```diff
@@ -2,21 +2,77 @@
 
 namespace Butler\Audit;
 
+use Butler\Audit\Bus\Dispatcher as BusDispatcher;
+use Butler\Audit\Bus\WithCorrelationId;
+use Butler\Audit\Facades\Auditor;
+use Illuminate\Bus\Dispatcher as BaseBusDispatcher;
 use Illuminate\Http\Client\PendingRequest;
+use Illuminate\Queue\Events\JobProcessed;
+use Illuminate\Queue\Events\JobProcessing;
+use Illuminate\Support\Facades\Queue;
 use Illuminate\Support\ServiceProvider as BaseServiceProvider;
-use Butler\Audit\Facades\Auditor;
 
 class ServiceProvider extends BaseServiceProvider
 {
     public function register()
     {
-        PendingRequest::macro('withCorrelationId', fn () => $this->withHeaders([
-            'X-Correlation-ID' => Auditor::correlationId(),
-        ]));
+        $this->addPendingRequestMacro();
+        $this->addDefaultInitiatorResolver();
+        $this->extendBusDispatcher();
     }
 
     public function boot()
     {
         $this->publishes([__DIR__ . '/../config/butler.php' => config_path('butler.php')], 'config');
+
+        $this->listenForJobProcessEvents();
+    }
+
+    private function addPendingRequestMacro(): void
+    {
+        PendingRequest::macro(
```

Comment:
Should we add something to the readme about this macro?

Reply:
Like this https://github.com/glesys/butler-audit#http-client-macro :wink: